### ISSUE TITLE
Inferred default parameter is erroneous using "typedef" "parameter"

### ISSUE THREAD

**Issue by cfischer** [palantir/tslint#1263]
### Bug Report

- __TSLint version__: `3.10.2`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: Visual Studio Code

#### TypeScript code being linted

```ts
class C {
  fn(str = '') {
  }
}
```
or
```ts
class C {
  fn(str: string = '') {
  }
}
```

with `tslint.json`:

```json
{
  "rules": {
    "typedef": [
      true,
      "parameter"
    ],
    "no-inferrable-types": true
  }
}
```

#### Actual behavior

You either get "LHS type (string) inferred by RHS expression, remove type annotation" or "expected parameter: 'str' to have a typedef"

#### Expected behavior

To not get "expected parameter: 'str' to have a typedef" if you specified a inferable default parameter value.

**Comment by JKillian** [comment]
Yes, this is an issue unfortunately. See https://github.com/palantir/tslint/issues/711 and #739
